tests/api/retrieval.test.js

```diff
@@ -0,0 +1,324 @@
+import { createMocks } from "node-mocks-http";
+import client from "next-auth/client";
+import retrieval from "../../pages/api/retrieval";
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import executeQuery from "@lib/integration/queryManager";
+import jwt from "jsonwebtoken";
+
+jest.mock("next-auth/client");
+jest.mock("@aws-sdk/client-dynamodb");
+jest.mock("@lib/integration/queryManager");
+
+jest.mock("@lib/integration/dbConnector", () => {
+  const mockClient = {
+    connect: jest.fn(),
+    query: jest.fn(),
+    end: jest.fn(),
+  };
+  return jest.fn(() => mockClient);
+});
+
+const dynamoClient = {
+  send: jest.fn(),
+};
+
+describe("/api/retrieval", () => {
+  beforeAll(() => {
+    process.env.TOKEN_SECRET = "gc-form-super-secret-code";
+  });
+  afterAll(() => {
+    delete process.env.TOKEN_SECRET;
+  });
+
+  it("Should return 400 status code b/c formID is undefined", async () => {
```

Comment:
`formID`, for this request as defined on Line 59, is an empty string, not `undefined`.

Reply:
Correct. Will update the comment and a new test to cover `undefined` case as well.